### PR TITLE
[PT-1.12.1][SM] re-release with the deprecated smdebug hooks removed from binary, fix PT-1.13.1 CVE checks

### PR DESCRIPTION
**Pull request by ShiboXing**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by ShiboXing** [comment]
1. all versions of future package has a DOS vulnerabilty, scanned by the safety report.
2. smdebugger needs a numpy deprecation fix

**Comment by ShiboXing** [comment]
efa: 6cf317a
standard: e978e6d
rc: 052e047

File: pytorch/training/docker/1.12/py3/cu113/Dockerfile.sagemaker.gpu.os_scan_allowlist.json [hunk 1]
```diff
@@ -50,11 +80,11 @@
...
-      "severity": "MEDIUM",
+      "source_url": "https://snyk.io/vuln/SNYK-PYTHON-TORCH-3149871",
+      "source": "SNYK",
+      "severity": "CRITICAL",
       "status": "ACTIVE",
-      "title": "CVE-2022-45907 - torch"
+      "title": "IN1-PYTHON-TORCH-3149871 - torch"
     }
   ]
 }
```
Comment (xncqr) [inline pytorch/training/docker/1.12/py3/cu113/Dockerfile.sagemaker.gpu.os_scan_allowlist.json:90]: Mind adding a new line under here for sanity? Notice the red circle on the end of the file?

File: pytorch/training/docker/1.13/py3/Dockerfile.ec2.cpu.os_scan_allowlist.json [hunk 2]
```diff
@@ -53,5 +53,35 @@
...
+      "cvss_v2_score": 0,
+      "cvss_v3_severity": "HIGH",
+      "source_url": "https://people.canonical.com/~ubuntu-security/cve/2022/CVE-2022-3996.html",
+      "source": "UBUNTU_CVE",
+      "severity": "LOW",
+      "status": "ACTIVE",
+      "title": "CVE-2022-3996 - cryptography"
+    }
+  ]
 }
```
Comment (xncqr) [inline pytorch/training/docker/1.13/py3/Dockerfile.ec2.cpu.os_scan_allowlist.json:87]: Mind adding a new line under here for sanity? Notice the red circle on the end of the file?

File: pytorch/training/docker/1.13/py3/Dockerfile.ec2.gpu.os_scan_allowlist.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      "cvss_v2_score": 0,
+      "cvss_v3_severity": "HIGH",
+      "source_url": "https://people.canonical.com/~ubuntu-security/cve/2022/CVE-2022-3996.html",
+      "source": "UBUNTU_CVE",
+      "severity": "LOW",
+      "status": "ACTIVE",
+      "title": "CVE-2022-3996 - cryptography"
+    }
+  ]
+}
```
Comment (xncqr) [inline pytorch/training/docker/1.13/py3/Dockerfile.ec2.gpu.os_scan_allowlist.json:32]: Mind adding a new line under here for sanity? Notice the red circle on the end of the file?

File: pytorch/training/docker/1.13/py3/cu117/Dockerfile.ec2.gpu.os_scan_allowlist.json [hunk 4]
```diff
@@ -1,57 +1,90 @@
...
+      "cvss_v2_score": 0,
+      "cvss_v3_severity": "CRITICAL",
+      "source_url": "https://people.canonical.com/~ubuntu-security/cve/2022/CVE-2022-45907.html",
+      "source": "UBUNTU_CVE",
+      "severity": "MEDIUM",
+      "status": "ACTIVE",
+      "title": "CVE-2022-45907 - torch"
+    }
+  ]
 }
```
Comment (xncqr) [inline pytorch/training/docker/1.13/py3/cu117/Dockerfile.ec2.gpu.os_scan_allowlist.json:90]: Mind adding a new line under here for sanity? Notice the red circle on the end of the file?

File: pytorch/training/docker/1.13/py3/cu117/Dockerfile.ec2.gpu.os_scan_allowlist.json [hunk 5]
```diff
@@ -1,57 +1,90 @@
...
+      "source_url": "https://people.canonical.com/~ubuntu-security/cve/2022/CVE-2022-3996.html",
+      "source": "UBUNTU_CVE",
+      "severity": "LOW",
+      "status": "ACTIVE",
+      "title": "CVE-2022-3996 - cryptography"
+    }
+  ],
+  "torch": [
+    {
+      "description": "[torch](https://pypi.org/project/torch) is a Tensors and Dynamic neural networks in Python with strong GPU acceleration\n\nAffected versions of this package are vulnerable to Command Injection in `torch.jit.annotations.parse_type_line` which can cause arbitrary code execution because `eval` is used unsafely.",
```
Comment (xncqr) [inline pytorch/training/docker/1.13/py3/cu117/Dockerfile.ec2.gpu.os_scan_allowlist.json:34]: we might want to add a "reason_to_ignore": to state why we are ignoring this Critical CVE. Example here -> https://github.com/xncqr/deep-learning-containers/blob/cf53bebe8d3fcf89259d68a90cc71912f93b71c0/pytorch/inference/docker/1.13/py3/Dockerfile.ec2.cpu.os_scan_allowlist.json#L4

One Block below would be the same.

File: pytorch/training/docker/1.13/py3/example/Dockerfile.gpu.example.os_scan_allowlist.json [hunk 6]
```diff
@@ -1,57 +1,90 @@
...
+      "source_url": "https://people.canonical.com/~ubuntu-security/cve/2022/CVE-2022-3996.html",
+      "source": "UBUNTU_CVE",
+      "severity": "LOW",
+      "status": "ACTIVE",
+      "title": "CVE-2022-3996 - cryptography"
+    }
+  ],
+  "torch": [
+    {
+      "description": "[torch](https://pypi.org/project/torch) is a Tensors and Dynamic neural networks in Python with strong GPU acceleration\n\nAffected versions of this package are vulnerable to Command Injection in `torch.jit.annotations.parse_type_line` which can cause arbitrary code execution because `eval` is used unsafely.",
```
Comment (xncqr) [inline pytorch/training/docker/1.13/py3/example/Dockerfile.gpu.example.os_scan_allowlist.json:34]: we might want to add a "reason_to_ignore": to state why we are ignoring this Critical CVE. Example here -> https://github.com/xncqr/deep-learning-containers/blob/cf53bebe8d3fcf89259d68a90cc71912f93b71c0/pytorch/inference/docker/1.13/py3/Dockerfile.ec2.cpu.os_scan_allowlist.json#L4

One Block below would be the same.

File: pytorch/training/docker/1.13/py3/example/Dockerfile.gpu.example.os_scan_allowlist.json [hunk 7]
```diff
@@ -1,57 +1,90 @@
...
+      "cvss_v2_score": 0,
+      "cvss_v3_severity": "CRITICAL",
+      "source_url": "https://people.canonical.com/~ubuntu-security/cve/2022/CVE-2022-45907.html",
+      "source": "UBUNTU_CVE",
+      "severity": "MEDIUM",
+      "status": "ACTIVE",
+      "title": "CVE-2022-45907 - torch"
+    }
+  ]
 }
```
Comment (xncqr) [inline pytorch/training/docker/1.13/py3/example/Dockerfile.gpu.example.os_scan_allowlist.json:90]: Mind adding a new line under here for sanity? Notice the red circle on the end of the file?

File: test/dlc_tests/container_tests/bin/pytorch_tests/setupPyTorchBackendTest [hunk 8]
```diff
@@ -10,10 +10,8 @@ sed -i '/from apex.fp16_utils import network_to_half, prep_param_lists, model_gr
 sed -i '/try:[ ]*$/{N;s/try:\s*from apex.parallel import DistributedDataParallel as DDP/from torch.nn.parallel import DistributedDataParallel as DDP/}' dlbs/python/pytorch_benchmarks/benchmarks.py
 sed -i 's/traceback.print_exc(file=sys.stdout)/traceback.print_exc(file=sys.stdout)\n        raise/g' dlbs/python/pytorch_benchmarks/benchmarks.py
 
-python -c "import sys;
-assert sys.version_info < (3, 7);"
-RETURN_VAL=`echo $?`
-if [ ${RETURN_VAL} -eq 1 ]; then
+RETURN_VAL=$(python -c "import sys; print(int(sys.version_info < (3, 7)));")
```
Comment (xncqr) [inline test/dlc_tests/container_tests/bin/pytorch_tests/setupPyTorchBackendTest:13]: Thank you for the update here.

File: pytorch/training/docker/1.12/py3/cu113/Dockerfile.sagemaker.gpu.os_scan_allowlist.json [hunk 9]
```diff
@@ -1,9 +1,39 @@
...
+      "status": "ACTIVE",
+      "title": "CVE-2022-3996 - cryptography"
+    }
+  ],
   "torch": [
     {
-      "description": "[torch](https://pypi.org/project/torch) is a Tensors and Dynamic neural networks in Python with strong GPU acceleration\n\nAffected versions of this package are vulnerable to Command Injection in `torch.jit.annotations.parse_type_line` which can cause arbitrary code execution because `eval` is used unsafely.",
-      "vulnerability_id": "SNYK-PYTHON-TORCH-3149871",
-      "name": "SNYK-PYTHON-TORCH-3149871",
+      "description": " In PyTorch before trunk/89695, torch.jit.annotations.parse_type_line can cause arbitrary code execution because eval is used unsafely.",
```
Comment (xncqr) [inline pytorch/training/docker/1.12/py3/cu113/Dockerfile.sagemaker.gpu.os_scan_allowlist.json:34]: we might want to add a "reason_to_ignore": to state why we are ignoring this Critical CVE. Example here -> https://github.com/xncqr/deep-learning-containers/blob/cf53bebe8d3fcf89259d68a90cc71912f93b71c0/pytorch/inference/docker/1.13/py3/Dockerfile.ec2.cpu.os_scan_allowlist.json#L4

One Block below would be the same.

File: pytorch/training/docker/1.12/py3/example/Dockerfile.gpu.example.os_scan_allowlist.json [hunk 10]
```diff
@@ -56,5 +56,37 @@
       "status": "ACTIVE",
       "title": "CVE-2022-45907 - torch"
     }
-  ]
+  ],
```
Comment (xncqr) [inline pytorch/training/docker/1.12/py3/example/Dockerfile.gpu.example.os_scan_allowlist.json:59]: Recommend adding the same "reason_for_ignore" to the 2 blocks above as they are listed as critical, though they are patched.

File: pytorch/buildspec-1-12.yml [hunk 11]
```diff
@@ -46,34 +46,34 @@ context:
...
+  #   tag_python_version: &TAG_PYTHON_VERSION py38
+  #   cuda_version: &CUDA_VERSION cu116
+  #   os_version: &OS_VERSION ubuntu20.04
+  #   tag: !join [ *VERSION, "-", *DEVICE_TYPE, "-", *TAG_PYTHON_VERSION, "-", *CUDA_VERSION, "-", *OS_VERSION, "-ec2" ]
+  #   docker_file: !join [ docker/, *SHORT_VERSION, /, *DOCKER_PYTHON_VERSION, /, *CUDA_VERSION, /Dockerfile.,
+  #                        *DEVICE_TYPE ]
+  #   target: ec2
+  #   context:
+  #     <<: *TRAINING_CONTEXT
   BuildSageMakerCPUPTTrainPy3DockerImage:
```
Comment (kenny-ezirim) [inline pytorch/buildspec-1-12.yml:77]: Are you releasing only SageMaker CPU image?

**Review by xncqr — APPROVED** [review]
See toml already reverted.. LGTM

File: pytorch/training/docker/1.13/py3/Dockerfile.ec2.gpu.os_scan_allowlist.json [hunk 12]
```diff
@@ -0,0 +1,33 @@
+{
```
Comment (kenny-ezirim) [inline pytorch/training/docker/1.13/py3/Dockerfile.ec2.gpu.os_scan_allowlist.json:1]: This file should be deleted. Allowlist json files for GPU should be under cu*** folders.

File: pytorch/training/docker/1.13/py3/cu117/Dockerfile.gpu [hunk 13]
```diff
@@ -169,7 +169,6 @@ RUN ompi_info --parsable --all | grep mpi_built_with_cuda_support:value \
     typing \
     h5py \
     requests \
-    future \
```
Comment (kenny-ezirim) [inline pytorch/training/docker/1.13/py3/cu117/Dockerfile.gpu:172]: Did you find a solution for addressing "future" package CVE other than adding the vulnerability id to ignore list?

File: test/dlc_tests/container_tests/bin/pytorch_tests/setupPyTorchBackendTest [hunk 14]
```diff
@@ -10,10 +10,8 @@ sed -i '/from apex.fp16_utils import network_to_half, prep_param_lists, model_gr
...
+if [ ${RETURN_VAL} -eq 0 ]; then
```
Comment (kenny-ezirim) [inline test/dlc_tests/container_tests/bin/pytorch_tests/setupPyTorchBackendTest:14]: The condition changed from equivalence to 1 to equivalence to 0. Can you confirm that the if statement code block should execute only when the system version is >= 3.7?

Comment (ShiboXing) [reply test/dlc_tests/container_tests/bin/pytorch_tests/setupPyTorchBackendTest:14]: yes this has been tested

File: pytorch/buildspec-1-12.yml (hunk 11 above)
Comment (ShiboXing) [reply pytorch/buildspec-1-12.yml:77]: SM CPU and GPU

File: pytorch/training/docker/1.13/py3/Dockerfile.ec2.gpu.os_scan_allowlist.json (hunk 12 above)
Comment (ShiboXing) [reply pytorch/training/docker/1.13/py3/Dockerfile.ec2.gpu.os_scan_allowlist.json:1]: acked, will fix it in PT-1.13 release PR

File: pytorch/training/docker/1.13/py3/cu117/Dockerfile.gpu (hunk 13 above)
Comment (ShiboXing) [reply pytorch/training/docker/1.13/py3/cu117/Dockerfile.gpu:172]: future package comes with python as module already, not need to install it
